ipatests/test_xmlrpc/tracker/user_plugin.py

```diff
@@ -62,22 +62,40 @@ class UserTracker(KerberosAliasMixin, Tracker):
 
     primary_keys = {u'uid', u'dn'}
 
-    def __init__(self, name, givenname, sn, **kwargs):
+    def __init__(self, name=None, givenname=None, sn=None, **kwargs):
+        """ Check for non-empty unicode string for the required attributes
+         in the init method """
+
+        if not isinstance(givenname, (str, unicode)) and len(givenname) > 0:
+            raise ValueError("No name provided: %s" % givenname)
```

Comment:
The condition does not fit the description, it should contain some more parentheses:
`if not (isinstance(givenname, (str, unicode)) and len(givenname) > 0):`
Also you can use 'basestring' instead of '(str, unicode)'.
Please change the error message so that it would unambiguously tell which value is incorrect (e.g. "No or invalid first name provided").